portfolio/src/main/java/com/google/sps/data/RequestJson.java

```diff
@@ -0,0 +1,240 @@
+    // This method calls the actual Book API from the URL and returns a JSON Object from the
+    // response convert space to url format
+
+    String url = String.format("https://www.googleapis.com/books/v1/volumes?q=%s", query);
+    // this link contains different data than the book's selfLink
+    URL obj = new URL(url);
+    HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+    con.setRequestMethod("GET");
+    con.setRequestProperty("User-Agent", "Mozilla/5.0");
+    int responseCode = con.getResponseCode();
+
+    BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+    String inputLine;
+    StringBuffer response = new StringBuffer(); // will contain the large string of information
+    while ((inputLine = in.readLine()) != null) {
+      response.append(inputLine);
+    }
+    in.close();
+
+    JSONObject myResponse = new JSONObject(response.toString());
+    return myResponse;
+  }
+
+  public ArrayList<FullBook> call_me(String query, int numberResults) throws Exception {
+    File index = new File("output");
+
+    if (!index.exists()) {
+      index.mkdir();
+    } else {
+      String[] entries = index.list();
+      for (String s : entries) {
+        File currentFile = new File(index.getPath(), s);
+        currentFile.delete();
+      }
+    }
+
+    ArrayList<FullBook> returnedList = new ArrayList<FullBook>();
+
+    JSONObject myResponse =
+        new JSONObject(requestQuery(query).toString()); // request Query returns a JSON object of
+                                                        // the JSON string with all the data
+    JSONArray allItems = myResponse.getJSONArray("items");
+
+    for (int i = 0; i < numberResults; i++) {
+      returnedList.add(jsonToFullBook(allItems.getJSONObject(i))); // jsonToFullBook
+    }
+    return returnedList;
+  }
+
+  public FullBook jsonToFullBook(JSONObject jsonObject) throws JSONException {
+    FullBook newBook = new FullBook();
+    // Map<String, String> tempISBN = new HashMap<String, String>();
+    String tempISBN = ""; // double nested [{}]
+
+    tempISBN = getIsbn(jsonObject).get("ISBN_13");
+    try {
+      newBook.setStringField("isbn", tempISBN);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    for (BookFieldsEnum field : BookFieldsEnum.values()) {
+      String jsProperty = field.getJSProperty(); // the string value in the enum
+      String tempProperty = "Undefined";
+      List<String> tempPropertyArray = new ArrayList<String>();
+      boolean isArray = false;
+
+      if (!jsProperty.equals("genre") && !jsProperty.equals("isbn")) {
+        try {
+          if (jsProperty.equals("categories") || jsProperty.equals("authors")) {
+            // seperate because they exists as arrays in the JSON object [] and genre has a
+            // different name ("categories")
+            isArray = true;
+            if (jsonObject.getJSONObject("volumeInfo").has(jsProperty)) {
```

Comment:
you seem to be using string literals like "volumeInfo" pretty often in this class. You can make the string literals into class private final constants to avoid typos, make changes easier, etc!